src/language/mongoDBService.ts

```diff
@@ -1,31 +1,33 @@
-import { CompletionItemKind, CancellationToken } from 'vscode-languageserver';
-import { Worker as WorkerThreads } from 'worker_threads';
-import { signatures } from '@mongosh/shell-api';
+/* eslint-disable no-sync */
 import * as util from 'util';
-import { Visitor } from './visitor';
-import { ServerCommands, PlaygroundRunParameters } from './serverCommands';
+import { CompletionItemKind, CancellationToken, Connection, CompletionItem } from 'vscode-languageserver';
+import fs from 'fs';
+import path from 'path';
+import { signatures } from '@mongosh/shell-api';
+import { Worker as WorkerThreads } from 'worker_threads';
 
-const path = require('path');
-const fs = require('fs');
+import { NodeOptions, ExecuteAllResult, ShellCompletionItem, CollectionItem } from '../utils/types';
+import { ServerCommands, PlaygroundRunParameters } from './serverCommands';
+import { Visitor } from './visitor';
 
 export const languageServerWorkerFileName = 'languageServerWorker.js';
 
 export default class MongoDBService {
-  _connection: any;
+  _connection: Connection;
   _connectionString?: string;
-  _connectionOptions?: any;
-  _cachedFields: object;
-  _cachedDatabases: [];
-  _cachedCollections: object;
-  _cachedShellSymbols: any;
+  _connectionOptions?: NodeOptions;
+  _cachedDatabases: CompletionItem[] | [];
```

Comment:
Maybe we can make this `_cachedDatabases: CompletionItem[] = []` and remove the setting in the constructor. Same with  `_cachedFields` and `_cachedCollections`